UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,234 @@
+     * @brief Returns direction
+     *
+     * @return direction
+     */
+    UNICORN_EXPORT glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Return right vector
+     *
+     * @return right vector
+     */
+    UNICORN_EXPORT glm::vec3 GetRight() const;
+
+    /**
+     * @brief Returns up vector
+     *
+     * @return up vector
+     */
+    UNICORN_EXPORT glm::vec3 GetUp() const;
+
+    /**
+     * @brief Return translate
+     *
+     * @return translate
+     */
+    UNICORN_EXPORT glm::vec3 GetTranslate() const;
+
+    /**
+     * @brief Returns model matrix
+     *
+     * @return model matrix
+     *
+     * Don't forget to recalculate @sa Calculate
+     */
+    UNICORN_EXPORT glm::mat4 const& GetModelMatrix() const;
+
+    /**
+     * @brief Translates by local X - right vector
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateLocalX(float distance);
+
+    /**
+     * @brief Translates by local Y - up vector
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateLocalY(float distance);
+
+    /**
+     * @brief Translates by local Z - direction vector
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateLocalZ(float distance);
+
+    /**
+     * @brief Translates by local X
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateWorldX(float distance);
+
+    /**
+     * @brief Translates by local Y
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateWorldY(float distance);
+
+    /**
+     * @brief Translates by local Z
+     *
+     * @param[in] distance distance to translate
+     */
+    UNICORN_EXPORT void TranslateWorldZ(float distance);
+
+    /**
+     * @brief Rotates by X axis counterclockwise
+     *
+     * @param[in] radians amount of radians
+     */
+    UNICORN_EXPORT void RotateX(float radians);
+
+    /**
+     * @brief Rotates by Y axis counterclockwise
+     *
+     * @param[in] radians amount of radians
+     */
+    UNICORN_EXPORT void RotateY(float radians);
+
+    /**
+     * @brief Rotates by Z axis counterclockwise
+     *
+     * @param[in] radians amount of radians
+     */
+    UNICORN_EXPORT void RotateZ(float radians);
+
+    /**
+     * @brief Rotates by all axises counterclockwise
+     *
+     * @param[in] rotation amount of radians at each axis
+     */
+    UNICORN_EXPORT void Rotate(glm::vec3 rotation);
+
+    /**
+     * @brief Rotates by quaternion
+     *
+     * @param[in] rotation quaternion
+     */
+    UNICORN_EXPORT void Rotate(glm::quat rotation);
+
+    /**
+     * @brief Rotates around point
+     *
+     * @param[in] radians amount of radians in frame
+     * @param[in] axis axis of rotation
+     * @param[in] point point of interest
+     */
+    UNICORN_EXPORT void RotateAroundPoint(float radians, glm::vec3 axis, glm::vec3 point);
```

Comment:
This feels like utility method that usercode shall use on its own, not via class interface